### ISSUE TITLE
Day02 ex05 bad example

### ISSUE THREAD

**Issue by gdufay** [42-AI/bootcamp_machine-learning#67]
* Day: 02
* Exercise: 05

In the gradient function, theta should be of dimension (n + 1) * 1, but in the example theta is of dimension n * 1

**Screenshots**
![image](https://user-images.githubusercontent.com/64479367/80581772-f0f4bb80-8a0d-11ea-85b0-cf3a45fa0ca9.png)

**Comment by madvid** [comment]
Plus, at the initialization of theta, shape is not correct as it shape is (n,) instead of (n,1).

**Comment by FlorianMuller** [comment]
To get a part of the output asked in the subject you need insert a `0` at the beginning of your thetas.

Something like that:
```python
theta1 = np.array([0, 3, 0.5, -6])
# [...]
theta2 = np.array([0, 0, 0, 0])
```

With those values, I get those output:
```python
# Exemple 1:
# me:      array([ -33.71428571,  -37.35714286, 183.14285714, -393.])
# subject: array([                -37.35714286, 183.14285714, -393.])

# Exemple 2:
# me:      array([ -0.71428571,   0.85714286, 23.28571429, -26.42857143])
# subject: array([                0.85714286, 23.28571429, -26.42857143])
```

Of course there's one more gradient because we have one more theta

**Comment by madvid** [comment]
Thanks for the issue, indeed there were a big mistake.
This issue is fixed in the commit : [dev 2f1b5c8]. It will appear on main branch at the next merge from dev into main branch.
I verified and took your examples @FlorianMuller, many thanks.
